src/sparse/merkle_tree.rs

```diff
@@ -0,0 +1,385 @@
+use crate::common::{AsPathIterator, Bytes32, Msb, Node as NodeTrait};
+use fuel_storage::Storage;
+
+use crate::sparse::hash::sum;
+use crate::sparse::{zero_sum, Buffer, Node, StorageNode};
+
+pub struct MerkleTree<'storage, StorageError> {
+    root_node: Node,
+    storage: &'storage mut dyn Storage<Bytes32, Buffer, Error = StorageError>,
+}
+
+impl<'a, 'storage, StorageError> MerkleTree<'storage, StorageError>
+where
+    StorageError: std::error::Error + Clone + 'static,
+{
+    pub fn new(
+        storage: &'storage mut dyn Storage<Bytes32, Buffer, Error = StorageError>,
+    ) -> Result<Self, Box<dyn std::error::Error>> {
+        let root_node = Node::create_placeholder();
+
+        storage.insert(&root_node.hash(), root_node.as_buffer())?;
+
+        Ok(Self { root_node, storage })
+    }
+
```

Comment:
Do we need to add a followup item to implement a `fn load()` for this type of tree?

Reply:
Yeah, that is certainly a desirable feature. I've got a ticket for it here: https://github.com/FuelLabs/fuel-merkle/issues/63. I have it separated like this so that I can (hopefully!) keep the PRs concise.